### ISSUE TITLE
Truncated text within flex containers breaks out rather than being contained/truncated

### ISSUE THREAD

**Issue by stefsullrew** [philipwalton/flexbugs#63]
I've found an interesting bug and a partial solution. I have text that is truncated inside a flex media object (and flex grids). In FFOX and IE, the truncated text doesn't truncate (Chrome, Safari and Edge work as expected). The truncated text takes up its full length, rudely smashing anything in its path. I created a minimum test case here: http://codepen.io/stefsullrew/pen/Kpyarb

I was able to fix the FFOX problem by adding min-width:0 to both the .col of my grid and the body of the media object (and I understand that's probably ok and Chrome is about to implement the same default now). My trouble remains with IE 11.

Philip, you'll see the code of your flexi media object in my Pen. My first problem was, I had no text showing at all (with the truncation class) in IE. I changed the `flex:1` on the media object's body to `flex: 1 1 auto;` and that was fixed. But IE still wants a pixel-based max-width on the truncated text, not a percentage-based, to truncate. Very aggravating since this needs to be responsive and my designer doesn't want to add a width. 

(I am able to truncate in IE in grids that have explicit percentage widths set on the columns.) I'd love to hear the thoughts of you wise flexperts here. :)

**Comment by gregwhitworth** [comment]
Started working on this, I think ultimately you're hitting some of the changes to min-width (Edge/FF/Canary has these), also you're hitting an interop issue between MS Edge/FF and Canary (somewhat similar to this [one](https://code.google.com/p/chromium/issues/detail?id=426898) but I can't be sure at this point). I have gotten it closer to working in [IE11](http://codepen.io/anon/pen/NqwgbX?editors=110), I need to get the ellipses working but this exposes the interop issue stated above.

In this use case you're trying to create, you'll need roughly three workarounds to make this work which is sub optimal, especially due to the complexity of flex and current changing in implementations (so you may break). My honest suggestion would be to not use flex for this *entire* use case, but try to simplify the nesting of the flexes.

I'll continue to try and find a work around that works in IE11+/FF/Chrome/Canary but there are so many interop issues between the four of those that it might be impossible :)

@philipwalton If you're able to take a glance as well, this will need a third pair of eyes and I would appreciate it as I'm pretty slammed ATM (thanks!!!)

**Comment by philipwalton** [comment]
So I think this is a genuine IE11 bug, and it appears to be fixed in Edge, so that's good. I'm not entirely sure if this is just one bug or a combination of things, so it's hard to know how to phrase it for others to avoid.

I think the Firefox issue you were noticing is actually Firefox doing it correctly because it implements the new `min-width: auto` part of the spec (Edge and Chrome Canary too). However, for your use-case, setting `min-width: 0` is probably what you want since a single line of text larger than the container would cause overflow.

As for solving your immediate problem, I think you can do that by approaching the layout from a different direction. Here's what you have currently:

- You have two columns, the left one has a computed flex value of `0 1 auto` and the right one `0 0 auto` (aka `none`).
- Neither column is growing, but they're spaced apart by setting `space-between` on the container.

From a layout perspective, this is basically the same as the following:

- You have two columns, the left one has a computed flex value of `1 1 0%` and the right one `0 0 auto` (aka `none`).
- When the content in the left column is sparse, there will still be space between the columns because the left column is set to grow.

I put together [this codepen](http://codepen.io/anon/pen/Xbzgzw?editors=110) using this second approach and it's working in FF/Chrome/IE11/Edge/Safari for me. I hope that helps!

p.s. I *really* wish codepen had support for diffing two pens, so you could easily see what I changed!

**Comment by stefsullrew** [comment]
Sorry, was on PTO yesterday for my birthday. Having a look now. :) And yes, I agree — after reading through the bugs and changes a couple days ago, FFOX seems to be doing it right and Chrome is about to implement the same it appears: https://code.google.com/p/chromium/issues/detail?id=426898

What I'm working on is an actual framework (that people will use in a variety of ways), so I'm trying to work things out without being too precious. If the rule needs to be "no flexbox in flexbox" that could be quite a frustration for the teams using this since grids are flexbox (and can be nested), media blocks are flexbox and horizontal lists are flexbox (with fallback for oldies). I may need to rethink and pull some of these items back into their older form. :sigh:

**Comment by gregwhitworth** [comment]
So @philipwalton solution that works cross browser won't work for this? In regards to flex updates, FF/Chrome still need to implement the abspos changes, but at that point Canary/Edge/Firefox will be at the same spot. The min-width change is actually in Canary now, but the interop issue you've found is a potential bug, but the overall feature is there. Let us know if your saying that the markup and layout has to remain as you set it up and we'll try to do what we can, but as stated there are bugs in these since they're so new and just now starting to be used which reveals edge cases which at times can be hard to test for.

**Comment by stefsullrew** [comment]
Thanks @gregwhitworth - I'm testing Philip's solution in the real component right now: http://codepen.io/anon/pen/Xbzgzw?editors=110

Since this was a highly simplified proof of concept, I have to test whether this solution will work in my actual components. I shall report back.

**Comment by stefsullrew** [comment]
It's not working. I'm making a new, more complex CodePen (since I boiled it down to the least code where the problem still showed up). Stay tuned.

**Comment by stefsullrew** [comment]
@gregwhitworth - The CodePen you shared has overlapping columns with no truncation at all in IE11: http://www.screencast.com/t/OuXtygYS6zn (And yes, with the `min-width: 0` I was able to solve my FFOX truncation — and I understand that other browsers will be making the same spec change.) 

However, nothing I do is solving the IE11 issue. I'm just going to take the media object out and inline-block the three items that need to stay together (which I believe will also get rid of a nested grid row). Like I said, what I'm building will be used by other people, so I can't get "too hacky"... Thanks for trying to assist though. Not sure exactly what the IE11 bug *is* but seems it should be called out? (I don't know that there's a workaround though.)

**Comment by gregwhitworth** [comment]
Completely agree, I'm not sure if there is a workaround either, it just seems like a straight up bug. Regarding my codepen, it wasn't finished, was just trying to get them to where they could at minimum overlap, I was going to try to figure out how to cause the truncation inside of the same markup, but was waiting to see if @philipwalton workaround would work. More than likely as you stated it may require too much hacking, especially for your purposes.

**Comment by philipwalton** [comment]
Ok, I'm going to close this for now. @stefsullrew feel free to reopen if you're able isolate your specific issue.

**Comment by stefsullrew** [comment]
Hey @philipwalton and @gregwhitworth - Even though this is closed, I'm going to leave the solution here for others. I've commented the new pen I forked from my first one. The solution was a couple of things. 

First, I took Philip's advice and changed from using justify-content: space-between:... to simply setting the second .col not to flex (which has the same effect).

Second, I reverted my .Media-body back to flex:1; as it had originally been, — but left the min-width:0; which has become important due to the changes in the flexbox spec.

Third, I added a class that we're finding very useful since we truncate loads of things and use a lot of flexbox. We call it .has-flexi-truncate, and it goes on the .col that contains the nested flexi-object that has the truncation inside (in this demo, it's the media object).

Hope this helps someone. I made notes in the new pen indicating what I changed since there's not a compare feature: http://codepen.io/stefsullrew/pen/WvWWqJ?editors=110
